shaman/util.py

```diff
@@ -155,3 +156,25 @@ def parse_distro_query(query):
             dict(distro=distro, distro_codename=codename, distro_version=version, arch=arch)
         )
     return result
+
+
+def get_repo_url(query, project_name, ref, arch, directory=None, repo_file=True):
+    # requires the repository to be fully available on a remote chacra
+    # instance for a proper redirect. Otherwise it will fail explicitly
+    if arch == 'aarch64':
```

Comment:
this is a bit naive because it is up to the client that uploads packages to chacra to define what the arch is. Nothing enforces what that really is. For example, this would break (or fail to match) if a client decided to use `AMD64` instead of `amd64`. 

I can't come up with a good alternative here either. The problem is that this is enforcing, and we don't enforce it anywhere else.